src/analysis/normed/ring/seminorm.lean

```diff
@@ -278,7 +281,123 @@ instance : has_one (mul_ring_norm R) :=
 
 instance : inhabited (mul_ring_norm R) := ⟨1⟩
 
+lemma neg_one_eq_one {R : Type*} [non_assoc_ring R] {f : mul_ring_norm R} :
+  f (-1) = 1 :=
+begin
+  have H₁ : f (-1) * f (-1) = 1,
+  calc
+    f (-1) * f (-1) = f ((-1) * (-1)) : by simp only [map_neg_eq_map, map_one, mul_one, mul_neg]
+    ...             = f 1 : by norm_num
+    ...             = 1 : f.map_one',
+  have H₂: f (-1) ≥ 0 := map_nonneg f (-1),
+  rw mul_self_eq_one_iff at H₁,
+  cases H₁,
+  { exact H₁ },
+  { rw H₁ at H₂,
+    have h' : ¬(-1 ≥ (0 : ℝ)) := by norm_num,
+    contradiction },
+end
+
+lemma pow_eq {R : Type*} [ring R] {f : mul_ring_norm R}
+  (r : R) (n : ℕ) : f (r ^ n) = (f r) ^ n :=
+begin
+  induction n with d hd,
+  { simp only [nat.nat_zero_eq_zero, pow_zero],
+    exact f.map_one' },
+  { rw [pow_succ, pow_succ, ← hd],
+    simp only [map_mul] }
+end
+
+lemma div_eq {R : Type*} [division_ring R] {f : mul_ring_norm R}
+  (p q : R) (hq : q ≠ 0) : f (p / q) = (f p) / (f q) :=
+begin
+  have H : f q ≠ 0,
+  { intro fq0,
+    exact hq (f.eq_zero_of_map_eq_zero' q fq0) },
+  calc f (p / q) = f (p / q) * f q / f q : by simp only [H, map_div₀, div_mul_cancel,
+                                                ne.def, not_false_iff]
+  ...            = f (p / q * q)  / f q : by simp only [map_mul]
+  ...            = f p / f q : by simp only [hq, div_mul_cancel, ne.def, not_false_iff]
+end
```

Comment:
This lemma is true by `rw map_div₀` which doesn't even use `hq`! So I think this can be removed